app/controllers/RulesController.scala

```diff
@@ -10,18 +11,14 @@ import scala.concurrent.{ExecutionContext, Future}
 /**
  * The controller that handles the management of validator rules.
  */
-class RulesController(cc: ControllerComponents, validatorPool: ValidatorPool, ruleResource: RuleResource, sheetId: String)(implicit ec: ExecutionContext)  extends AbstractController(cc) {
+class RulesController(cc: ControllerComponents, validatorPool: ValidatorPool, languageToolFactory: LanguageToolFactory, ruleResource: RuleResource, sheetId: String)(implicit ec: ExecutionContext)  extends AbstractController(cc) {
   def refresh = Action.async { implicit request: Request[AnyContent] =>
     for {
       (rulesByCategory, ruleErrors) <- ruleResource.fetchRulesByCategory()
-      errorsByCategory <- Future.sequence(
-        rulesByCategory.map { case (category, rules) => {
-          validatorPool.updateConfig(category.name, ValidatorConfig(rules))
-        }}.toList
-      )
+      errorsByCategory = addValidatorToPool(rulesByCategory)
       rules <- validatorPool.getCurrentRules
     } yield {
-      val errors = errorsByCategory.flatten ::: ruleErrors
+      val errors = errorsByCategory.map(_._2).flatten ::: ruleErrors
```

Comment:
Not a big deal but could be `errorsByCategory.flatMap(_._2)`

Reply:
👍